### PR TITLE
Add Chrome backend

### PR DESCRIPTION
**Pull request by NiR-**

### CODE REVIEW

File: src/Knp/Snappy/Chrome/AbstractGenerator.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+namespace Knp\Snappy\Chrome;
+
+use Knp\Snappy\Filesystem;
+use Knp\Snappy\Generator;
+use Knp\Snappy\LocalGenerator;
+
+/**
+ * Abstract chrome generator.
+ *
+ * @author Albin Kerouanton <albin.kerouanton@knplabs.com<
```
Comment (Einenlum) [inline src/Knp/Snappy/Chrome/AbstractGenerator.php]: >

File: src/Knp/Snappy/Chrome/Image.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Knp\Snappy\Chrome;
+
+/**
+ * Image generator based on chrome.
+ *
+ * @author Albin Kerouanton <albin.kerouanton@knplabs.com<
```
Comment (Einenlum) [inline src/Knp/Snappy/Chrome/Image.php]: \>

File: src/Knp/Snappy/Chrome/Pdf.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Knp\Snappy\Chrome;
+
+/**
+ * PDF generator based on chrome.
+ *
+ * @author Albin Kerouanton <albin.kerouanton@knplabs.com<
```
Comment (Einenlum) [inline src/Knp/Snappy/Chrome/Pdf.php]: \>

File: src/Knp/Snappy/Chrome/AbstractGenerator.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate($input, string $output, array $options = [], bool $overwrite = false)
+    {
+        $this->filesystem->prepareOutput($output, $overwrite);
+
+        $this->doGenerate(sprintf('file://%s', $input), $output, $options);
```
Comment (barryvdh) [inline src/Knp/Snappy/Chrome/AbstractGenerator.php]: It should only use file:// when it's a local file, not an URL

File: src/Knp/Snappy/Chrome/AbstractGenerator.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOutputFromHtml($html, array $options = [])
+    {
+        $temporaryFile = $this->filesystem->createTemporaryFile(null, $this->getDefaultExtension());
+
+        $this->doGenerate(sprintf('data:text/html,%s', rawurlencode($html)), $temporaryFile, $options);
```
Comment (barryvdh) [inline src/Knp/Snappy/Chrome/AbstractGenerator.php]: Is there an advantage in doing this over a temp file? Performance doesn't seem to have a big impact here. Or would it be faster in slow-disk systems? It does have some limitations like 2MB limit (which is large for HTML, but still).

Also, rawurlencode doesn't seem to be required?

**Pull request merged**